include/rocksdb/cloud/cloud_env_options.h

```diff
@@ -530,9 +530,14 @@ class CloudEnv : public Env {
   // Files both in S3 and in the local directory have this [epoch] suffix.
   virtual std::string RemapFilename(const std::string& logical_name) const = 0;
 
-  // return the list of live files under bucket/object path
-  virtual Status FindAllLiveFiles(const std::string& bucket,
-                                  const std::string& object_path,
+  // Fetch current Manifest file into local
+  virtual Status FetchCurrentManifest(const std::string& local_dbname) const = 0;
```

Comment:
Do we need this in the public API?

Reply:
yeah. This is how I plan to use it:
```
cloud_env->PreloadCloudManifest(local_dbname);  // fetch cloud manifest to local and load it into cloud_env
cloud_env->FetchCurrentManifest(local_dbname); // fetch current cloud manifest
cloud_env->FindLiveFiles(local_dbname, &file_nums, &manifest); // find live files based on local manifest
```
But we can probably merge `FetchCurrentManifest` and `FindLiveFiles` into one

Reply:
Check out my other comment.

Reply:
are you worried that misusage of this API could cause issues?

How about I removed `FetchCurrentManifest`, and in `FindLiveFiles`, we ask caller to provide a temporary filename for the fetched Manifest.

```
Status FindLiveFiles(const std::string& local_dbname, const std::string& manifest_tmp_fname, &file_nums, &manifest);
```
In this function, we fetch the current Manifest into `manifest_tmp_fname`

Reply:
Have you considered just using `NewSequentialFile()` on the manifest filename, which will both download the manifest locally and open it? I think it's the simplest solution and is also what we use in the regular codepath, so it's well-tested.

Public APIs once introduced need to be documented, maintained and it's very hard to change or remove them.